apps/backend/src/evaluations/evaluations.controller.ts

```diff
@@ -72,6 +70,13 @@ export class EvaluationsController {
       abac.can(Action.Read, evaluation)
     );
 
+    // Ensure user has the ability to read associated groups
+    evaluations.forEach((evaluation) => {
+      evaluation.groups = evaluation.groups.filter((group) =>
+        abac.can(Action.Read, group)
+      );
+    });
+
     return evaluations.map(
       (evaluation) =>
         new EvaluationDto(evaluation, abac.can(Action.Update, evaluation))
```

Comment:
May be able to save a loop over all of the data by performing the filtering in the map below instead of a separate foreach loop.